api/include/opentelemetry/context/runtime_context.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "opentelemetry/context/context.h"
+#include "opentelemetry/context/context_config.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+class RuntimeContext
+{
+public:
+  // Return the current context.
+  static Context *GetCurrent() { return context_handler_.GetCurrent(); }
+
+  // Sets the current 'Context' object. Returns a token
+  // that can be used to reset to the previous Context.
+  static ContextToken Attach(Context *context) { return context_handler_.Attach(context); }
+
+  // Resets the context to a previous value stored in the
+  // passed in token. Returns true if successful, false otherwise
+  static bool Detach(ContextToken &token) { return context_handler_.Detach(token); }
+
+  static ContextType context_handler_;
+};
+
+ContextType RuntimeContext::context_handler_ = ContextType();
```

Comment:
This needs to be an interface that ThreadLocalContext implements. It *could* be RuntimeContext if it also had a couple pure virtual methods.

It feels like the Token type needs to be defined as part of that interface, and derived classes could access GetContext either by making it public on Token, or by having a concrete protected visibility accessor in  like
Context* GetTokenContext(Token& t)
in the base interface.

Reply:
The Token is now a part of the RuntimeContext and the ThreadLocalContext is now derived from RuntimeContext again.